File: app/src/main/java/bassamalim/hidaya/activities/WelcomeActivity.java
package bassamalim.hidaya.activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceManager;

import bassamalim.hidaya.R;
import bassamalim.hidaya.databinding.ActivityWelcomeBinding;

public class WelcomeActivity extends AppCompatActivity {

    private ActivityWelcomeBinding binding;

    public WelcomeActivity() {}    // added because of crashes without it

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityWelcomeBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        if (savedInstanceState == null)
            getSupportFragmentManager().beginTransaction().replace(R.id.settings,
                    new Settings.SettingsFragment("initial")).commit();

        setListeners();
    }

    private void setListeners() {
        binding.saveBtn.setOnClickListener(v -> {
            binding.settings.setVisibility(View.GONE);
            binding.saveBtn.setVisibility(View.GONE);
            binding.disclaimerSpace.setVisibility(View.VISIBLE);

            SharedPreferences.Editor editor = PreferenceManager
                    .getDefaultSharedPreferences(this).edit();
            editor.putBoolean("new_user", false);
            editor.apply();
        });

        binding.agreed.setOnClickListener(view -> {
            Intent intent = new Intent(this, Splash.class);
            startActivity(intent);
            finish();
        });
        binding.rejected.setOnClickListener(view -> launchAnyWay());
    }

    private void launchAnyWay() {
        Intent intent = new Intent(this, MainActivity.class);
        intent.putExtra("located", false);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/bassamalim/hidaya/activities/WelcomeActivity.kt
package bassamalim.hidaya.activities

import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager
import bassamalim.hidaya.R
import bassamalim.hidaya.databinding.ActivityWelcomeBinding
import bassamalim.hidaya.fragments.LocationFragment
import bassamalim.hidaya.fragments.SettingsFragment
import bassamalim.hidaya.other.Utils

class WelcomeActivity: AppCompatActivity() {

    private lateinit var binding: ActivityWelcomeBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Utils.myOnActivityCreated(this)
        binding = ActivityWelcomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        if (savedInstanceState == null)
            supportFragmentManager.beginTransaction().replace(
                R.id.frame, SettingsFragment.newInstance(true)
            ).commit()

        setListeners()
    }

    private fun setListeners() {
        binding.saveBtn.setOnClickListener {
            binding.saveBtn.visibility = View.GONE

            supportFragmentManager.beginTransaction().replace(
                R.id.frame, LocationFragment.newInstance("initial")
            ).commit()

            val editor: SharedPreferences.Editor =
                PreferenceManager.getDefaultSharedPreferences(this).edit()
            editor.putBoolean("new_user", false)
            editor.apply()
        }
    }

}